### PR TITLE
Fix high cpu load due to recurring prune job

### PR DESCRIPTION
**Pull request by BjoernAtBosch**
## Describe your changes

In class `GrpcClient` a `RecurringJob` is used to clean-up completed gRPC requests. This leads to permanent job execution as the recurring job always re-adds itself to the `ThreadPool` and then is exceuted again and again.

This fix replaces the recurring job by pruning possible completed requests (from the `m_activeCalls` list) when adding new calls/requests.

Furthermore, it fixes the busy wait in the `VehicleApp::run()` method.

## Issue ticket number and link

none

## Checklist - Manual tasks

* [ ] Examples are executing successfully
* [ ] Created/updated unit tests. Code Coverage percentage on new code shall be >= 80%.
* [ ] Created/updated integration tests.

* [ ] Devcontainer can be opened successfully
* [ ] Devcontainer can be opened successfully behind a corporate proxy
* [ ] Devcontainer can be re-built successfully

* [ ] Extended the documentation (e.g. README.md, CONTRIBUTING.md, Velocitas Docs)

### CODE REVIEW

File: examples/seat-adjuster/src/SeatAdjusterApp.cpp [hunk 1]
```diff
@@ -126,8 +127,17 @@ void SeatAdjusterApp::onErrorTopic(const velocitas::Status& status) {
 }
 } // namespace example
 
+std::unique_ptr<example::SeatAdjusterApp> myApp;
+
+void signal_handler(int sig) {
```
Comment (doosuu) [inline examples/seat-adjuster/src/SeatAdjusterApp.cpp:132]: We should think about moving this to a common place in the SDK, maybe introducing a `Launcher` which spawns the app registers the signal handler?

e.g.
```cpp
template<typename TAppClass> 
class Launcher {
public:
   Launcher() = default;
   ~Launcher() = default;

  void signalHandler() { ... }

   template<typename TArgs...>
   void launch(TArgs... args) {
      m_app = std::make_unique<TAppClass>(...args);
      signal(SIGINT, &Launcher<TAppClass>::signalHandler);
   }


private:
   std::unique_ptr<TAppClass> m_app;
}
```

File: sdk/src/sdk/grpc/GrpcClient.cpp [hunk 2]
```diff
@@ -15,19 +15,15 @@
...
-    m_recurringJob = std::make_shared<RecurringJob>([this]() { pruneCompletedRequests(); });
-    ThreadPool::getInstance()->enqueue(m_recurringJob);
-}
+void GrpcClient::addActiveCall(std::shared_ptr<GrpcCall> call) {
+    pruneCompletedRequests();
 
-GrpcClient::~GrpcClient() {
-    m_recurringJob->cancel();
-    m_recurringJob->waitForTermination();
+    std::scoped_lock<std::mutex> lock(m_mutex);
```
Comment (doosuu) [inline sdk/src/sdk/grpc/GrpcClient.cpp:25]: lock should be scoped with an inner block

**Review by doosuu — CHANGES_REQUESTED** [review]

File: sdk/src/sdk/grpc/BrokerAsyncGrpcFacade.h [hunk 3]
```diff
@@ -22,11 +22,16 @@
 
 #include "sdv/databroker/v1/broker.grpc.pb.h"
 
+#include <functional>
```
Comment (doosuu) [inline sdk/src/sdk/grpc/BrokerAsyncGrpcFacade.h:25]: Are these added headers really needed?

File: examples/seat-adjuster/src/SeatAdjusterApp.cpp (hunk 1 above)
Comment (doosuu) [reply examples/seat-adjuster/src/SeatAdjusterApp.cpp:132]: Shall we create a ticket for this?

Comment (BjoernAtBosch) [reply examples/seat-adjuster/src/SeatAdjusterApp.cpp:132]: Issue created: #90

**Review by doosuu — APPROVED** [review]
lgtm

**Pull request merged**